operator/src/main/java/oracle/kubernetes/operator/helpers/ResponseStep.java

```diff
@@ -92,17 +94,34 @@ private NextAction getPotentialRetryAction(Packet packet) {
 
   /**
    * Returns next action that can be used to get the next batch of results from a list search that
-   * specified a "continue" value.
+   * specified a "continue" value, if any; otherwise, returns next.
    *
+   * @param callResponse Call response
    * @param packet Packet
    * @return Next action for list continue
    */
-  protected final NextAction doContinueList(Packet packet) {
-    RetryStrategy retryStrategy = packet.getSpi(RetryStrategy.class);
-    if (retryStrategy != null) {
-      retryStrategy.reset();
+  protected final NextAction doContinueListOrNext(CallResponse<T> callResponse, Packet packet) {
+    return doContinueListOrNext(callResponse, packet, getNext());
+  }
+
```

Comment:
Here you have the retry logic and list continue logic mixed together. It seems to me that they really need to be separate, as a retry and and a continue are orthogonal. You seem to be conditioning the processing of a retry on the presence of a continue token. At the very least, the code is unclear.

Reply:
I've changed this to try and clarify... I believe that the disconnect is that I see that you implemented a solution in your PR that would iterate until there was no continue token and then present the final list to the response step, but I wanted to present each partial list to the response step. Therefore, the response step needs a mechanism to request the next request. When a new request happens, I do want to reset any failure count that may have been accumulated from the previous request.